cpp/src/io/utilities/column_utils.cuh

```diff
@@ -67,13 +67,17 @@ rmm::device_uvector<column_device_view> create_leaf_column_device_views(
       size_type index) mutable {
       col_desc[index].parent_column = parent_col_view.begin() + index;
       column_device_view col        = parent_col_view.column(index);
-      if (col_desc[index].stats_dtype != dtype_byte_array) {
-        // traverse till leaf column
-        while (col.type().id() == type_id::LIST or col.type().id() == type_id::STRUCT) {
-          col = (col.type().id() == type_id::LIST)
-                  ? col.child(lists_column_view::child_column_index)
-                  : col.child(0);
+      // traverse till leaf column
+      while (col.type().id() == type_id::LIST or col.type().id() == type_id::STRUCT) {
+        auto child = (col.type().id() == type_id::LIST)
+                       ? col.child(lists_column_view::child_column_index)
+                       : col.child(0);
+        // stop early if writing a byte array, it needs to be a list<int8> not the int8 column
+        if (col_desc[index].stats_dtype == dtype_byte_array &&
+            (child.type().id() == type_id::INT8 || child.type().id() == type_id::UINT8)) {
```

Comment:
Does Spark differentiate binary types stored as int8 and uint8? Are both of these equivalent from Spark’s perspective? Can Spark normalize inputs to use only one or the other? It feels very suspicious to me that we’ve decided to accept either one across the binary code paths.

Reply:
`std::byte` is neither signed nor unsigned. https://godbolt.org/z/bo1rorxGM

Interpreting a `std::byte` as a signed or unsigned requires an explicit cast. We can probably pick one and stick with it.

Reply:
Cpp reference lists it as: `enum class byte : unsigned char {} ;`, which is why I said that. I see your point though and it should be agnostic as it is simply a byte. I would personally pick uint8 for the column type, but it seems that int8 has momentum due to its use in string columns. Happy to go either way. Should I make a new issue to address this?

Reply:
Yes, a new issue sounds good. I'd suggest using `int8` to align with strings columns. Moreover, tracking the signed-ness is not relevant to us, since there's no arithmetic operations needed on binary data and certain optimizations may be easier for signed types.